src/adapters/http/index.ts

```diff
@@ -0,0 +1,109 @@
+import Adapter from '../../lib/adapter.js';
+import GleeMessage from '../../lib/message.js';
+import http from 'http';
+import { validateData } from '../../lib/util.js';
+import GleeError from '../../errors/glee-error.js';
+import * as url from 'url';
+
+class HttpAdapter extends Adapter {
+    private res: any
+
+    name(): string {
+        return 'HTTP server'
+    }
+
+    async connect(): Promise<this> {
+        return this._connect();
+    }
+
+    async send(message: GleeMessage): Promise<void> {
+        console.log('control reached send');
+        return this._send(message);
+    }
+
+    _connect(): Promise<this> {
+        return new Promise((resolve, reject) => {
+            const serverUrl = new URL(this.serverUrlExpanded);
+            const httpServer = this.glee.options?.websocket?.httpServer || http.createServer();
```

Comment:
We shouldn't use the HTTP server from the WebSocket configuration. That's only for WebSocket servers. You should use something like this instead:

```suggestion
            const httpServer = this.glee.options?.http?.server || http.createServer();
```